src/main/resources/templates/testRules.html

```diff
@@ -18,23 +18,27 @@ <h1 id="test_rules_header">Test Rules</h1>
         </div>
         <div class="row px-3">
             <div class="p-1 col-lg-6 col-md-6 col-12 border border-right border-success" id="testRulesDOMObject">
-                <div class="d-flex flex-wrap">
+                <div class="d-flex flex-wrap sticky-button-bar">
 	                <div class="col-11">
-	                   <button class="btn btn-primary upload_rules mr-1 mt-1 d-inline-block">
-                           Load Rules from file
+	                   <button class="btn btn-primary add_rule rule-button mr-1 mt-1 d-inline-block" data-bind="click : $root.addRule.bind($data, ruleTemplate)">
+                           Add
+                           <i class="fa fa-fw  fa-plus"></i>
+                       </button>
+	                   <button class="btn btn-primary upload_rules rule-button mr-1 mt-1 d-inline-block">
+                           Load From File
                            <i class="fa fa-fw  fa-upload"></i>
                        </button>
                        <input class="hide" type='file' id='uploadRulesFile' name='file'/>
+                       <button class="btn btn-primary download_rules mr-1 mt-1 d-inline-block">
+                           Download
+                           <i class="fa fa-fw  fa-download"></i>
+                       </button>
 	                   <button class="btn btn-primary download_rules_template rule-button mr-1 mt-1 d-inline-block">
                            Get Template
                            <i class="fa fa-fw  fa-download"></i>
                        </button>
-	                   <button class="btn btn-primary add_rule rule-button mr-1 mt-1 d-inline-block" data-bind="click : $root.addRule.bind($data, ruleTemplate)">
-                           Add Rule
-                           <i class="fa fa-fw  fa-plus"></i>
-                       </button>
-                       <button class="btn btn-danger clear_all_rules rule-button mr-1 mt-1 d-inline-block" data-bind="click : $root.clearAllRules.bind()">
-                           Clear All Rules
+                       <button class="btn btn-danger clear_all_rules rule-button mr-1 mt-1 d-inline-block" data-bind="click : $root.confirmClearAll.bind($data, 'rules')">
```

Comment:
Maybe the rule-button can class can be removed now since both sides have the exact same text inside the buttons now. The uniform buttons looked better in responsive design as the screen got smaller.